miniwin/src/internal/d3drmrenderer.h

```diff
@@ -0,0 +1,19 @@
+#pragma once
+
+#include "miniwin/d3drm.h"
+
+#include <SDL3/SDL.h>
+
+typedef struct PositionColorVertex {
+	float x, y, z;
+	float nx, ny, nz;
+	Uint8 r, g, b, a;
+} PositionColorVertex;
+
+class Direct3DRMRenderer {
+public:
+	virtual void SetBackbuffer(SDL_Surface* backbuffer) = 0;
+	virtual void PushVertices(const PositionColorVertex* vertices, size_t count) = 0;
+	virtual void SetProjection(D3DRMMATRIX4D perspective, D3DVALUE front, D3DVALUE back) = 0;
+	virtual HRESULT Render() = 0;
+};
```

Comment:
This API does not allow the SDL3GPU renderer to buffer meshes on the GPU.
Soon, we'll also need to buffer textures on the GPU.

Reply:
Yeah it's super rudimentary. The thinking is that the API will change as the implementation grows. But with the fallback software render there it also makes sure that the SDL3GPU implementation doesn't pollut the generic parts and that the interface doesn't become to hostile to alternate backends (at the same time the software render being internal allows for good flexibility in what can be changed).

Reply:
I'm thinking a lot of my D3DRM code will slowly migrate to the software renderer as more of it gets implemented on the GPU side ... I just don't really know where the limits are in that regard :sweat_smile: